lib/services/sf.js

```diff
@@ -485,9 +485,8 @@ function SnowflakeService(connectionConfig, httpClient, config)
     operationQueue.length = 0;
   };
 
-  this.isConnected = function ()
-  {
-    return currentState === stateConnected;
+  this.isConnected = function () {
+    return currentState === stateConnected || currentState === stateRenewing;
```

Comment:
It is correct. I'm still wondering if state 'Connetcing' should be added too? WDYT?

Reply:
It looks like `stateConnecting` occurs from `statePristine`. And `statePristine` occurs when there's no tokens like the first time connect is called or after the connection is terminated

So it's possible the user checks if the connection is valid while it's already connecting. And if they try to manually connect while it's in the connecting state, they'll get an error as well so I think we can add `stateConnecting` to prevent that